### PR TITLE
[audio_to_spectrogram] Enable publishing frequency vs amplitude plot

### PR DESCRIPTION
**Pull request by iory**
# What is this?

This PR enables publishing frequency vs amplitude plot images in `spectrum_plot.py`.

image

### CODE REVIEW

File: audio_to_spectrogram/scripts/spectrum_plot.py [hunk 1]
```diff
@@ -35,10 +57,16 @@ def _cb(self, msg):
         self.ax.set_xlim((self.freq.min(), self.freq.max()))
         self.ax.set_ylim((0.0, 20))
         self.ax.legend(loc='upper right')
+        print(self.pub_img.get_num_connections())
```
Comment (708yamaguchi) [inline audio_to_spectrogram/scripts/spectrum_plot.py:60]: You may forget to remove this line?

If you need this line, `rospy.loginfo` or `rospy.logdebug` seems better.

File: audio_to_spectrogram/scripts/spectrum_plot.py [hunk 2]
```diff
@@ -35,10 +57,16 @@ def _cb(self, msg):
...
+        if self.pub_img.get_num_connections() > 0:
```
Comment (708yamaguchi) [inline audio_to_spectrogram/scripts/spectrum_plot.py:61]: How about using `ConnectionBasedTransport`?
https://jsk-docs.readthedocs.io/projects/jsk_common/en/latest/jsk_topic_tools/class/python_connection_based_transport.html

**Review by 708yamaguchi — COMMENTED** [review]
Thank you very much for the nice feature!

I think it would be helpful to publish spectrum plot as `sensor_msgs/Image`.

Please consider about my reviews.

File: audio_to_spectrogram/scripts/spectrum_plot.py (hunk 1 above)
Comment (iory) [reply audio_to_spectrogram/scripts/spectrum_plot.py]: Thanks. I forgot to remove the debug print.
Fixed.

File: audio_to_spectrogram/scripts/spectrum_plot.py (hunk 2 above)
Comment (iory) [reply audio_to_spectrogram/scripts/spectrum_plot.py:67]: That's good idea, thanks. I modified the code. https://github.com/jsk-ros-pkg/jsk_recognition/pull/2654/commits/2e86235c63a0d80d6497db22606e8940ef1bbaf2

File: audio_to_spectrogram/scripts/spectrum_plot.py [hunk 3]
```diff
@@ -23,10 +45,17 @@ def __init__(self):
...
-        # ROS subscriber
+        # ROS publisher and subscriber
+        self.pub_img = self.advertise(
+            '~output/viz', sensor_msgs.msg.Image, queue_size=1)
+
+    def subscribe(self):
         self.sub_spectrum = rospy.Subscriber(
             '~spectrum', Spectrum, self._cb, queue_size=1000)
 
+    def unsupported(self):
```
Comment (708yamaguchi) [inline audio_to_spectrogram/scripts/spectrum_plot.py:56]: Typo: unsubscribe?

```
Traceback (most recent call last):
  File "/home/leus/ros/melodic/src/jsk-ros-pkg/jsk_recognition/audio_to_spectrogram/scripts/spectrum_plot.py", line 77, in <module>
    SpectrumPlot()
  File "/home/leus/ros/melodic/src/jsk-ros-pkg/jsk_common/jsk_topic_tools/src/jsk_topic_tools/transport.py", line 26, in __call__
    obj = type.__call__(cls, *args, **kwargs)
TypeError: Can't instantiate abstract class SpectrumPlot with abstract methods unsubscribe
```

Comment (iory) [reply audio_to_spectrogram/scripts/spectrum_plot.py]: Sorry, I was wrong. Fixed. https://github.com/jsk-ros-pkg/jsk_recognition/pull/2654/commits/4fa0c14ab74111cf08731dda46c1d27cbc5ad3d0

**Comment by 708yamaguchi** [comment]
I am sorry for the many comments, but the last thing.

How about moving these lines to outside of `<group if="$(arg gui)">`?
https://github.com/jsk-ros-pkg/jsk_recognition/blob/6ff031d51d425bf77456d89fbe84b61e58da3aea/audio_to_spectrogram/launch/audio_to_spectrogram.launch#L56-L58

This is because the new `spectrum_plot.py` does not create GUI window.

**Comment by iory** [comment]
> I am sorry for the many comments, but the last thing.

Don't worry about it. That's a good idea. Modified https://github.com/jsk-ros-pkg/jsk_recognition/pull/2654/commits/6cd009c5fb99c3250f0f6e2bc65f72164ee53a5b

**Comment by 708yamaguchi** [comment]
Thank you so much!

I think this PR is OK to be merged after tests pass.

**Review by 708yamaguchi — APPROVED** [review]

**Comment by k-okada** [comment]
@iory please add plotted graph image on this comment thread or document

**Comment by iory** [comment]
I added the image in the README.md.
https://github.com/jsk-ros-pkg/jsk_recognition/blob/c57662c8b2ef7bb921447e8f683125b7db646172/audio_to_spectrogram/README.md#spectrum_plotpy

**Pull request merged**